src/grid/atomgrid.py

```diff
@@ -92,52 +90,48 @@ def __init__(
             )
         self._rot = rotate
         # check degs and size
-        if degs is None:
-            if not isinstance(size, (np.ndarray, list)):
-                raise TypeError(f"size is not type: np.array or list, got {type(size)}")
-            degs = convert_lebedev_sizes_to_degrees(size)
-        if not isinstance(degs, (np.ndarray, list)):
-            raise TypeError(f"degs is not type: np.array or list, got {type(degs)}")
-        if len(degs) == 1:
-            degs = np.ones(rgrid.size, dtype=int) * degs
-        self._rad_degs = degs
+        if degrees is None:
+            if not isinstance(sizes, (np.ndarray, list)):
+                raise TypeError(
+                    f"sizes is not type: np.array or list, got {type(sizes)}"
+                )
+            degrees = convert_lebedev_sizes_to_degrees(sizes)
+        if not isinstance(degrees, (np.ndarray, list)):
+            raise TypeError(
+                f"degrees is not type: np.array or list, got {type(degrees)}"
+            )
+        if len(degrees) == 1:
+            degrees = np.ones(rgrid.size, dtype=int) * degrees
+        self._degs = degrees
         self._points, self._weights, self._indices = self._generate_atomic_grid(
-            self._rgrid, self._rad_degs, rotate=self._rot
+            self._rgrid, self._degs, rotate=self._rot
         )
         self._size = self._weights.size
 
     @classmethod
-    def from_predefined(
-        cls,
-        atnum,
-        rgrid,
-        grid_type,
-        *_,
-        center=None,
-        rotate=False,
-    ):
+    def from_predefined(cls, rgrid, atnum, preset, *_, center=None, rotate=False):
```

Comment:
May I know the reason to swap the position of `radial grid` and `atomic number`

Reply:
All constructors take `rgrid` as an argument, so like the other two, I thought `from_predefined` should also have it as the first argument. I think it's important to be consistent. I would be happy to change it back if you feel strongly about it.

Reply:
One of the reason I prefer `atomic number` first, because logically, I think it's what pop up naturally in my head if I'm going to generate a grid for certain atom. I'd think I'm gonna generate a grid for some atom, then I would wonder what radial grid should I use to best fit the atom.

The inconsistency comes from the fact that only the `predefined` is `atomic number` sensitive. I got how you feel about the first args changing-inconsistency, but I'd think logic matters more. I will think about how to make it feels better

Reply:
Or we can make it a key-arg at the end of the function call. I am fine with that.

Reply:
Which argument are you suggesting to turn into a keyword argument? I am a bit confused.

Let's see what others have to say. I mostly wanted to make sure we are making a conscious decision, and this choice isn't an artifact of code legacy. This is a minor issue after all. 

I understand your point. But, from the ease-of-use perspective, I think the user should not need to remember which constructor takes `rgrid` as the 1st argument and which constructor takes `rgrid` as the 2nd argument. Even though 

The `from_pruned` also takes `atomic radius` as an argument, so based on your logic, should that also come first?

Reply:
Let's make `atnum` as a key-arg then it should be good.

For `from_pruned`, I don't remember the radius of any atoms except H. so it's fairly strange to me to think it in the first place.

Reply:
I'd tend to prefer the same order of arguments, so I guess starting with rgrid. I see Derrick's point, but insofar as rgrid is essential always (and a bit unexpected, at least for me) it's good to start with it. Also, I can imagine a way to use a default atnum if we needed to...I'm not recommending that, but it would be more sensible than a universal choice of radial grid.